### PR TITLE
Implement blackstone and deepslate blocks and walls

### PR DESCRIPTION
**Pull request by TwistedAsylumMC**

### CODE REVIEW

File: server/block/blackstone.go [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+	drops := oneOf(b)
+	if b.Type == GildedBlackstone() {
+		drops = func(item.Tool, []item.Enchantment) []item.Stack {
+			if rand.Float64() < 0.1 {
+				return []item.Stack{item.NewStack(item.GoldNugget{}, rand.Intn(4)+2)}
+			}
+			return []item.Stack{item.NewStack(b, 1)}
+		}
+	}
+	return newBreakInfo(hardness, pickaxeHarvestable, pickaxeEffective, drops).withBlastResistance(30)
```
Comment (DaPigGuy) [inline server/block/blackstone.go:31]:
```suggestion
	drops := oneOf(b)
	if b.Type == GildedBlackstone() {
		drops = func(item.Tool, []item.Enchantment) []item.Stack {
			if rand.Float64() < 0.1 {
				return []item.Stack{item.NewStack(item.GoldNugget{}, rand.Intn(4)+2)}
			}
			return []item.Stack{item.NewStack(b, 1)}
		}
	}
	return newBreakInfo(1.5, pickaxeHarvestable, pickaxeEffective, drops).withBlastResistance(30)
```

File: server/block/deepslate.go [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+type Deepslate struct {
+	solid
+	bassDrum
+
+	// Type is the type of deepslate of the block.
+	Type DeepslateType
+	// Axis is the axis which the deepslate faces.
+	Axis cube.Axis
+}
+
```
Comment (DaPigGuy) [inline server/block/deepslate.go:20]: Missing smelt info for cobbled deepslate

File: server/block/deepslate_bricks.go [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+// DeepslateBricks are a brick variant of deepslate and can spawn in ancient cities.
+type DeepslateBricks struct {
+	solid
+	bassDrum
+
+	// Cracked specifies if the deepslate bricks is its cracked variant.
+	Cracked bool
+}
+
```
Comment (DaPigGuy) [inline server/block/deepslate_bricks.go:11]: Missing smelt info

File: server/block/deepslate_tiles.go [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+// DeepslateTiles are a tiled variant of deepslate and can spawn in ancient cities.
+type DeepslateTiles struct {
+	solid
+	bassDrum
+
+	// Cracked specifies if the deepslate tiles is its cracked variant.
+	Cracked bool
+}
+
```
Comment (DaPigGuy) [inline server/block/deepslate_tiles.go:11]: Missing smelt info

File: server/block/polished_blackstone_brick.go [hunk 5]
```diff
@@ -0,0 +1,32 @@
+package block
+
+type PolishedBlackstoneBrick struct {
```
Comment (DaPigGuy) [inline server/block/polished_blackstone_brick.go:3]: Missing doc

**Review by DaPigGuy — CHANGES_REQUESTED** [review]

**Review by JustTalDevelops — APPROVED** [review]

**Comment by Sandertv** [comment]
I imagine it's easiest if we merge these after slabs and stairs and add those in this PR.

File: server/block/blackstone.go [hunk 6]
```diff
@@ -0,0 +1,49 @@
+package block
+
+import (
+	"github.com/df-mc/dragonfly/server/item"
+	"github.com/df-mc/dragonfly/server/world"
+	"math/rand"
+)
+
+// Blackstone is a naturally generating block in the nether that can be used to craft stone tools, brewing stands and
+// furnances. Gilded blackstone also has a 10% chance to drop 2-6 golden nuggets.
```
Comment (Sandertv) [inline server/block/blackstone.go:10]: Typo here, `furnances`.

**Review by Sandertv — COMMENTED** [review]
One typo, should be good otherwise when stairs/slabs are added. Feel free to merge once that's resolved.

**Pull request merged**